### DISCUSSION TITLE
Incorrect dimensions after image is downloaded

### DISCUSSION THREAD

**Discussion by Nevinia210** [pnp/pnpcore#953] [Q&A] [answered]
Hello Everyone,
After i load image with method getImageByServerRelativeUrl i receive image with vice versa height and width

```C#
var file = Program.context.Web.GetFileByServerRelativeUrl(imageUrl);
byte[] downloadedContentBytes = file.GetContentBytes();
Bitmap bmp;
using (var ms = new MemoryStream(downloadedContentBytes))
 {
        bmp = new Bitmap(ms);
        int originalHeight = bmp.Height;
        int originalWidth = bmp.Width;
 }
```
Real dimensions of an image are:
Width: 2252px
Height: 2984px

and after executing this code I'm getting
originalHeight: 2252px
originalWidth: 2984px

Could you confirm that this is a bug or did I do something wrong?

**Comment by jansenbe** [comment, marked as answer]
@Nevinia210 : I think what you see is expected, the code you're using is simply pulling down the file as is from SharePoint. You can try downloading the file using the UI and verify if you see the same

**Reply by Nevinia210** [reply]
@jansenbe : thank you for your comments, in the end i find out that it is really not issue in pnp, but rather iphone,
according to this article [stackoverflow](https://stackoverflow.com/questions/6455979/how-to-get-the-the-dimensions-of-an-image-file) iphone images are automatically rotated to landscape,
this code fix that issue:
```C#
using (var imageFactory = new ImageFactory())
{
    imageFactory
        .Load(stream)
        .AutoRotate(); //takes care of ex-if
    var height = imageFactory.Image.Height,
    var width = imageFactory.Image.Width
}
```

**Reply by jansenbe** [reply]
Thanks for sharing the fix. Are you using PnP.Core in an iOS app @Nevinia210 ?

**Reply by Nevinia210** [reply]
@jansenbe - luckily no, from what i read about ImageFactory, it has support only for older .net that runs only on windows, currently im using this library on .net5.0 and im planning to use it only on windows machines - so far it works as it should
